source/extensions/filters/udp/udp_proxy/hash_policy_impl.cc

```diff
@@ -0,0 +1,38 @@
+#include "extensions/filters/udp/udp_proxy/hash_policy_impl.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace UdpProxy {
+
+class SourceIpHashMethod : public HashPolicyImpl::HashMethod {
+public:
+  uint64_t evaluate(const Network::Address::Instance& downstream_addr) const override {
+    ASSERT(downstream_addr.ip() != nullptr);
```

Comment:
The `ip()` cannot be null because the implementation is here : `const Ip* ip() const override { return &ip_; }`.
So, I added the assert instead of null checking.

Reply:
Don't we support unix domain sockets for UDP? If not, this seems fragile as we might eventually? I think I would make this a normal check?

/wait-any

Reply:
Oh.. Sorry. I missed that. :(
I will revert it and add more test to prevent such mistakes. :)

Reply:
Now, it is fixed.